### ISSUE TITLE
Feature: support sort in findOrCreate method

### ISSUE THREAD

**Issue by elennaro** [balderdashy/waterline#1025]
It'd be great.

**Comment by aars** [comment]
Should be, though it could depend on the adapter used: https://github.com/balderdashy/waterline-docs/blob/master/query-methods.md#findorcreate-search-criteria-values-callback-

**Comment by dmarcelino** [comment]
@elennaro, does it have to be a `sort` or just some predicable order? As in, the records would be returned in the same order as submitted, for example:
``` javascript
model.findOrCreate([{ val: 1 }, { val: 2 }, { val: 3 }], function(err, records){
  console.log(records);  // prints [{ val: 1 }, { val: 2 }, { val: 3 }]
});
``` 

For adapters that don't implement their own `findORCreate`, the second option should be a simple change in the shim similar to #980 / #795.

**Comment by elennaro** [comment]
Thank you for your answer @dmarcelino.
Yes, I supposed to get values  sorted according to the query criteria I use.
I have:
models/Workdays.js
````js
module.exports = {
	attributes: {
		id: { type: "date", primaryKey: true },
		.....
//Relations
		.....
	}
};
````
and query in controllers/WorkdayController.js
````js
module.exports = {
	find: function(req, res) {
		var week = []; //some logic that generates 7 workday sequentially;
		Workday.findOrCreate({ where: { id: week.map(function(d) {
				return d.id;
			})}, limit: 7, sort: 'id DESC' }, week).exec(function(err, workdays) {
			//I have had hope to see sorted workdays here...
		});
	}
};
````

As you can see, it understands the query Criteria in this form:
````
{ where: { id: week.map(function(d) {
				return d.id;
			})}, limit: 7, sort: 'id DESC' }
````
But it ignores sort. Maybe it works as it should work, then it could be great to add such note to documentation.
I use sails-mongo with waterline. If sails-mongo overrrides findOrCreate then it's probably a sails-mongo issue.
Or it's not an issue at all...

**Comment by dmarcelino** [comment]
I see, that's a nifty code example :)

If all the records exist I would say this should be easy, the problem I see is if some or even all records don't exist. In the latter case waterline would have to do an extra `.find()` (or use [waterline-criteria](https://github.com/balderdashy/waterline-criteria)) to be able to sort all records...

**Comment by elennaro** [comment]
Wow waterline-criteria looks promising.
To be honest there is no problem to do a custom sort in callback after all.
So did I.

Reviewing the sources makes me understand, that custom sort is the easiest solution in this case, if I want to have database adapter independent code, and we both do not want to do extra db queries in the request.

So do you want me to close the issue?
And sorry for taking your time, I thought it could be a great feature, but I see now, that it's better to learn from sources before post issues.

**Comment by dmarcelino** [comment]
waterline-criteria is already being used in waterline so maybe it wouldn't be too difficult to sort the records coming from `findOrCreateEach` if a `sort` is present in the criteria. It wouldn't have any impact for those not trying to sort the results.

> So do you want me to close the issue?

Leave it for now, let's see what others says or if someone volunteers a patch!

**Comment by sailsbot** [comment]
Thanks for posting, @elennaro.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
